### DISCUSSION TITLE
Is it possible to completely disable close confirmation?

### DISCUSSION THREAD

**Discussion by mkpaz** [wezterm/wezterm#2780] [Q&A] [answered]
I tried this, but still get the close confirmation on tab middle click.

```lua
window_close_confirmation = 'NeverPrompt',
skip_close_confirmation_for_processes_named = { 'bash', 'sh', 'zsh', 'fish', 'tmux' },
keys = {
  {
    key = 'w', mods = 'CTRL|SHIFT', action = wezterm.action.CloseCurrentTab { confirm = false }
  },
}
```

It looks like `skip_close_confirmation_for_processes_named` not working for me, while I don't use multiplexer and don't have any spawned processes, just a pure bash shell. I spent like an hour trying to fix this with no luck and let me say it's very annoying to not be able to disable something that is default in any other terminal emulator. Also, requiring to redefine keymap to disable the confirmation looks like dirty hack, it's really should be less complicated.

**Comment by wez** [comment]
I'd like to understand more about why `skip_close_confirmation_for_processed_named = {...}` isn't working for you. I think there's more to your setup than you've shared and that may be influencing it.
Please file a bug report issue for this, and fill out the information requested by the template!

**Comment by mkpaz** [comment, marked as answer]
Digging into this a bit more I found out that this is flatpak issue. Tried to grant every possible permission via Flatseal with no luck and switched to rpm package. Both rpm and appimage don't have this problem. I only picked Flatpak because wezterm isn't in Fedora repos thus it lacks auto-updates if installed other way.

**Reply by wez** [reply]
Interesting. I opened https://github.com/wez/wezterm/issues/2783 to track this.

**Comment by joel-bluedata** [comment]
It also seems like ssh and unix domains have a different behavior here than the local domain. I have this in my config, trying various things to skip the tab close confirmation (config.keys already exists when this code is run):
```lua
  config.window_close_confirmation = 'NeverPrompt'
  table.insert(
    config.keys,
    {
      key = 'w',
      mods = 'SHIFT|CTRL',
      action = wezterm.action.CloseCurrentTab { confirm = false },
    }
  )
  wezterm.on('mux-is-process-stateful', function(_proc)
    return false
  end)
```

ctrl-shift-W works as expected, with no confirmation. In the local domain, I can also close tabs without confirmation when using the "x" on the tab. In an ssh or unix domain however, using the "x" on the tab still shows the confirmation dialog. (I can file an issue if I'm not just overlooking something here.)
